molgenis-api-data/src/main/java/org/molgenis/api/data/v3/DataServiceV3Impl.java

```diff
@@ -110,6 +121,58 @@ public void delete(String entityTypeId, String entityId) {
     repository.deleteById(typedEntityId);
   }
 
+  @Transactional
+  @Override
+  public void delete(String entityTypeId, Query query) {
+    Repository<Entity> repo =
+        metaDataService
+            .getRepository(entityTypeId)
+            .orElseThrow(() -> new UnknownEntityTypeException(entityTypeId));
+    org.molgenis.data.Query molgenisQuery = queryMapperV3.map(query, repo);
+    repo.delete(molgenisQuery.findAll());
+  }
+
+  @Override
+  public List<Entity> find(
+      String entityTypeId,
+      Query q,
+      Sort sort,
+      Selection filter,
+      Selection expand,
+      int size,
+      int number) {
+    Repository<Entity> repo =
+        metaDataService
+            .getRepository(entityTypeId)
+            .orElseThrow(() -> new UnknownEntityTypeException(entityTypeId));
+
+    Fetch fetch = toFetch(repo.getEntityType(), filter, expand);
+
+    org.molgenis.data.Query<Entity> query = queryMapperV3.map(q, repo);
+    query.fetch(fetch);
+    query.offset(number * size);
+    query.pageSize(size);
+
+    query.sort(map(sort));
+
+    return query.findAll().collect(Collectors.toList());
+  }
+
+  @Override
+  public int count(String entityTypeId, Query q) {
+    QueryV3Mapper queryV3Mapper = new QueryV3Mapper();
+    Repository<Entity> repo =
```

Comment:
Should throw a UnknownRepositoryException since the entity type might exist but be abstract. In that case replace with a call to ```private Repository<Entity> getRepository(String entityTypeId)```.